compiler-core/src/parse.rs

```diff
@@ -1882,23 +1882,28 @@ where
         let return_annotation = self.parse_type_annotation(&Token::RArrow)?;
 
         let (body, end, end_position) = match self.maybe_one(&Token::LeftBrace) {
-            Some(_) => {
+            Some((left_brace_start, _)) => {
                 let some_body = self.parse_statement_seq()?;
-                let (_, rbr_e) = self.expect_one(&Token::RightBrace)?;
+                let (_, right_brace_end) = self.expect_one(&Token::RightBrace)?;
                 let end = return_annotation
                     .as_ref()
                     .map(|l| l.location().end)
                     .unwrap_or(rpar_e);
                 let body = match some_body {
                     None => vec1![Statement::Expression(UntypedExpr::Todo {
-                        kind: TodoKind::EmptyFunction,
-                        location: SrcSpan { start, end },
+                        kind: TodoKind::EmptyFunction {
+                            function_location: SrcSpan { start, end }
+                        },
+                        location: SrcSpan {
+                            start: left_brace_start + 1,
+                            end: right_brace_end
+                        },
```

Comment:
Previously when the compiler inserted an implicit `todo` in an empty function body it marked its position as starting at the start of the function declaration. This would mess up code actions that rely on a function's body's statement to always come inside the curly braces.

I found this out implementing the "Destructure argument" code action and changed it here so that an implicit todo now has the following span:
```gleam
   pub fn main() { }
//                ^^ This span inside the function's body
// ^^^^^^^^^^^^^ Instead of this span outside the function's body
```
It still carries around the entire function span to produce nice error messages for incomplete code in the `TodoKind::EmptyFunction` variant